### PR TITLE
OSOE-482: Enforce parameter splatting instead of backtick in Lombiq.Analyzers.PowerShell

### PR DESCRIPTION
**Pull request by BenedekFarkas**
[OSOE-482](https://lombiq.atlassian.net/browse/OSOE-482)

### CODE REVIEW

File: .github/actions/verify-submodule-pull-request/Check-Parent.ps1 [hunk 1]
```diff
@@ -4,7 +4,7 @@
 )
 
 $url = "https://api.github.com/repos/$Repository/pulls?state=open&per_page=100"
-$titles = curl -s -H 'Accept: application/vnd.github.v3+json' $url | ConvertFrom-Json | ForEach-Object { $PSItem.title }
+$titles = Invoke-WebRequest -s -H 'Accept: application/vnd.github.v3+json' $url | ConvertFrom-Json | ForEach-Object { $PSItem.title }
```
Comment (Piedone) [inline .github/actions/verify-submodule-pull-request/Check-Parent.ps1]: Is `Invoke-WebRequest` really such a drop-in replacement to curl? Also point `validate-this-pull-request` to `issue/OSOE-482` so this is tested.

File: .github/actions/build-dotnet/Build-DotNetSolutionOrProject.ps1 [hunk 2]
```diff
@@ -51,7 +51,7 @@ if (Test-Path src/Utilities/Lombiq.Gulp.Extensions/Lombiq.Gulp.Extensions.csproj
 # This prepares the solution or project with the Lombiq.Analyzers files. The output and exit code are discarded because
 # they will be in error if there is a project without the LombiqNetAnalyzers target. Then there is nothing to do, and
 # the target will still run on the projects that have it.
-dotnet msbuild '-target:Restore;LombiqNetAnalyzers' $SolutionOrProject | Out-Null || bash -c 'true'
+dotnet msbuild '-target:Restore;LombiqNetAnalyzers' $SolutionOrProject | Out-Null -or bash -c 'true'
```
Comment (Piedone) [inline .github/actions/build-dotnet/Build-DotNetSolutionOrProject.ps1]: Also point all the affected action calls in `build-and-test-orchard-core` to the issue branch so these script changes are tested.

File: .github/actions/create-jira-issues-for-community-activities/Initialize-IssueDetails.ps1 [hunk 3]
```diff
@@ -8,8 +8,8 @@ param
     $PullReqestJiraIssueDescription
 )
 
-$context = [string]::IsNullOrEmpty($IssueComponent) ? $GitHub.repository : $IssueComponent
-$titleSuffix = $SuffixIssueTitles ? " in $context" : ""
+$context = [string]::IsNullOrEmpty($IssueComponent) | Where-Object $GitHub.repository : $IssueComponent
+$titleSuffix = $SuffixIssueTitles | Where-Object " in $context" : ""
```
Comment (Piedone) [inline .github/actions/create-jira-issues-for-community-activities/Initialize-IssueDetails.ps1]: These are ternary operators.

File: .github/actions/create-jira-issues-for-community-activities/Initialize-IssueDetails.ps1 [hunk 4]
```diff
@@ -54,7 +54,7 @@ switch ($GitHub.event_name)
     default
     {
         $message = "Unknown event `"$($GitHub.event_name)`". Please only call this script for one of the following " +
-            "events: discussion, issues, pull_request."
+        "events: discussion, issues, pull_request."
```
Comment (Piedone) [inline .github/actions/create-jira-issues-for-community-activities/Initialize-IssueDetails.ps1]: The original indentation was good, it's a continuation of the previous string.

File: .github/actions/publish-nuget/New-NuGetPackage.ps1 [hunk 5]
```diff
@@ -16,7 +16,7 @@
 
 param([array] $Arguments)
 
-$projects = (Test-Path *.sln) ? (dotnet sln list | Select-Object -Skip 2 | Get-Item) : (Get-ChildItem *.csproj)
+$projects = (Test-Path *.sln) | Where-Object (dotnet sln list | Select-Object -Skip 2 | Get-Item) : (Get-ChildItem *.csproj)
```
Comment (Piedone) [inline .github/actions/publish-nuget/New-NuGetPackage.ps1]: This is a ternary operator.

File: .github/actions/publish-nuget/Update-ManifestVersion.ps1 [hunk 6]
```diff
@@ -9,7 +9,8 @@ $manifests = Get-ChildItem $WorkDir -File -Recurse -Filter "Manifest.cs" |
     Select-String -List -Pattern '(OrchardCore.Modules.Manifest|OrchardCore.DisplayManagement.Manifest)' |
     Select-Object -ExpandProperty Path
 
-foreach ($manifest in $manifests) {
+foreach ($manifest in $manifests)
+{
```
Comment (Piedone) [inline .github/actions/publish-nuget/Update-ManifestVersion.ps1:13]: Do we also have an analyzer for this now?

File: .github/actions/setup-sql-server/Initialize-SqlServer.ps1 [hunk 7]
```diff
@@ -12,7 +12,7 @@ else
         '--detach', 'mcr.microsoft.com/mssql/server:2019-latest'
     )
 
-    docker pull mcr.microsoft.com/mssql/server &&
-    docker run @dockerRunSwitches &&
+    docker pull mcr.microsoft.com/mssql/server
+    docker run @dockerRunSwitches
```
Comment (Piedone) [inline .github/actions/setup-sql-server/Initialize-SqlServer.ps1]: Why did you remove these? Aren't these working across lines?

File: .github/actions/test-dotnet/Invoke-SolutionTests.ps1 [hunk 8]
```diff
@@ -14,12 +14,12 @@ $ConnectionStringSuffix = ";MultipleActiveResultSets=True;Connection Timeout=60;
 if ($Env:RUNNER_OS -eq "Windows")
 {
     $Env:Lombiq_Tests_UI__SqlServerDatabaseConfiguration__ConnectionStringTemplate =
-        "Server=.\SQLEXPRESS;Database=LombiqUITestingToolbox_{{id}};Integrated Security=True" + $ConnectionStringSuffix
+    "Server=.\SQLEXPRESS;Database=LombiqUITestingToolbox_{{id}};Integrated Security=True" + $ConnectionStringSuffix
```
Comment (Piedone) [inline .github/actions/test-dotnet/Invoke-SolutionTests.ps1]: Here and on line 22 the original indentation was good.

File: .github/actions/test-dotnet/Invoke-SolutionTests.ps1 [hunk 9]
```diff
@@ -60,8 +61,8 @@ foreach ($test in $tests) {
         # This is for xUnit ITestOutputHelper, see https://xunit.net/docs/capturing-output.
         '--logger', 'console;verbosity=detailed'
         '--verbosity', $Verbosity
-        $Filter ? '--filter' : ''
-        $Filter ? $Filter : ''
+        $Filter | Where-Object '--filter' : ''
+        $Filter | Where-Object $Filter : ''
```
Comment (Piedone) [inline .github/actions/test-dotnet/Invoke-SolutionTests.ps1]: Ternary operators.

File: .github/actions/test-dotnet/Merge-FailureDumps.ps1 [hunk 10]
```diff
@@ -3,7 +3,7 @@
 New-Item -Type Directory "$Directory/FailureDumps"
 
 $testDirectory = "$Directory/test"
-$rootDirectory = (Test-Path -Path $testDirectory) ? $testDirectory : $Directory
+$rootDirectory = (Test-Path -Path $testDirectory) | Where-Object $testDirectory : $Directory
```
Comment (Piedone) [inline .github/actions/test-dotnet/Merge-FailureDumps.ps1]: This is a ternary oeprator.

File: .github/actions/test-dotnet/Set-XUnitMaxParallelThreads.ps1 [hunk 11]
```diff
@@ -1,19 +1,19 @@
...
-        Filter = "xunit.runner.json"
-        Recurse = $true
-        # -FollowSymLink is needed because without it errors will be thrown of type:
-        # "Get-ChildItem: The system cannot find the path specified."
-        FollowSymlink = $true
-        # Ignore warnings that will be printed due to multiple symlinks pointing to the same physical file.
-        WarningAction = "Ignore"
+    # -Filter is the fastest way to find files because its value is passed directly to the file system API.
+    Filter        = "xunit.runner.json"
+    Recurse       = $true
```
Comment (Piedone) [inline .github/actions/test-dotnet/Set-XUnitMaxParallelThreads.ps1]: Such tab-arranged assignments are not something we do anywhere, so let's not start it in just this script.

**Review by Piedone — CHANGES_REQUESTED** [review]

File: .github/actions/verify-submodule-pull-request/Check-Parent.ps1 (hunk 1 above)
Comment (sarahelsaig) [reply .github/actions/verify-submodule-pull-request/Check-Parent.ps1]: It is not. None of the parameters are interoperable.

Comment (Piedone) [reply .github/actions/verify-submodule-pull-request/Check-Parent.ps1]: Just using curl here is fine BTW, `Invoke-WebRequest` isn't better.

File: .github/actions/test-dotnet/Set-XUnitMaxParallelThreads.ps1 (hunk 11 above)
Comment (BenedekFarkas) [reply .github/actions/test-dotnet/Set-XUnitMaxParallelThreads.ps1]: > I'd prefer not to do this tab stop-alignment here either.

Agreed, for reference, I replied on another PR: https://github.com/Lombiq/Utility-Scripts/pull/28#discussion_r1051637023

File: .github/actions/build-dotnet/Build-DotNetSolutionOrProject.ps1 (hunk 2 above)
Comment (BenedekFarkas) [reply .github/actions/build-dotnet/Build-DotNetSolutionOrProject.ps1]: Revisiting this now, what's the expected output here? Both of those (one the two sides of the `|`) commands return a boolean value and they are just ORed together?

Btw Windows PowerShell doesn't accept this and I'm still getting analyzer errors locally, but the code analysis passes in GHA. I'll undo the changes that have some uncertainty in them or has a potential for side effects and see what the code analysis says in GHA.

File: .github/actions/create-jira-issues-for-community-activities/Initialize-IssueDetails.ps1 (hunk 3 above)
Comment (BenedekFarkas) [reply .github/actions/create-jira-issues-for-community-activities/Initialize-IssueDetails.ps1]: Yeah, I didn't notice these - Invoke-Analyzers modified these with the -Fix switch (when calling from Windows PowerShell, because it doesn't support ternary operators).

File: .github/actions/create-jira-issues-for-community-activities/Initialize-IssueDetails.ps1 (hunk 4 above)
Comment (BenedekFarkas) [reply .github/actions/create-jira-issues-for-community-activities/Initialize-IssueDetails.ps1]: I agree. VS Code autoformatting changed this, but I don't know why. If you put brackets around it, then the indentation is correct, but I'll just revert this for now.

File: .github/actions/publish-nuget/New-NuGetPackage.ps1 (hunk 5 above)
Comment (BenedekFarkas) [reply .github/actions/publish-nuget/New-NuGetPackage.ps1]: Same as above.

File: .github/actions/publish-nuget/Update-ManifestVersion.ps1 (hunk 6 above)
Comment (BenedekFarkas) [reply .github/actions/publish-nuget/Update-ManifestVersion.ps1:13]: VS Code autoformatting fixes this.

File: .github/actions/setup-sql-server/Initialize-SqlServer.ps1 (hunk 7 above)
Comment (BenedekFarkas) [reply .github/actions/setup-sql-server/Initialize-SqlServer.ps1]: Also incompatible with Windows PowerShell (it looked like a mistake, reverting).

File: .github/actions/test-dotnet/Invoke-SolutionTests.ps1 (hunk 8 above)
Comment (BenedekFarkas) [reply .github/actions/test-dotnet/Invoke-SolutionTests.ps1]: Autoformatting again, although it's a different case here.

File: .github/actions/verify-submodule-pull-request/Check-Parent.ps1 (hunk 1 above)
Comment (BenedekFarkas) [reply .github/actions/verify-submodule-pull-request/Check-Parent.ps1]: Invoke-Analyzers with the Fix switch replaced this, because curl is considered an alias, so that's not correct, because the parameters should be changed.
Invoke-WebRequest would be better, because we can expect that it does the same thing (AFAIK because the curl available on Windows is not compiled from the same code as what's available on Linux, but pwsh is the same cross-platform code). We also used Invoke-WebRequest already in other repos/scripts instead of curl.

File: .github/actions/publish-nuget/Update-ManifestVersion.ps1 (hunk 6 above)
Comment (Piedone) [reply .github/actions/publish-nuget/Update-ManifestVersion.ps1:13]: Could you please do a quick check if there's a built-in or community analyzer for such formatting in PSAnalyzer? I'd be surprised to see a built-in one, because we should've noticed it originally, but for such basics things even a community one would be great.

File: .github/actions/build-dotnet/Build-DotNetSolutionOrProject.ps1 (hunk 2 above)
Comment (Piedone) [reply .github/actions/build-dotnet/Build-DotNetSolutionOrProject.ps1]: The double-pipe is not an OR operator: https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_pipeline_chain_operators?view=powershell-7.3#long-description However, I'm not clear on why we need this bash trick. @DAud-IcI you've written this originally in 89118d3b8e11af2cc5216f532b349f20cddf64f9. Is this to not fail the script if the `dotnet msbuild` command fails ("exit code are discarded")?

File: .github/actions/verify-submodule-pull-request/Check-Parent.ps1 (hunk 1 above)
Comment (Piedone) [reply .github/actions/verify-submodule-pull-request/Check-Parent.ps1]: Let's see, but please make sure to test this script's action/workflow too.

Comment (BenedekFarkas) [reply .github/actions/verify-submodule-pull-request/Check-Parent.ps1]: Will do.

File: .github/actions/publish-nuget/Update-ManifestVersion.ps1 (hunk 6 above)
Comment (BenedekFarkas) [reply .github/actions/publish-nuget/Update-ManifestVersion.ps1:13]: I actually checked the rest of the rules (not enabled by default), now why sure didn't notice them (I was focusing on getting rid of backticks at the time though), but yes, there are: https://github.com/Lombiq/PowerShell-Analyzers/issues/16

File: .github/actions/verify-submodule-pull-request/Check-Parent.ps1 [hunk 12]
```diff
@@ -3,12 +3,17 @@
...
 
-$url = "https://api.github.com/repos/$Repository/pulls?state=open&per_page=100"
-$titles = curl -s -H 'Accept: application/vnd.github.v3+json' $url | ConvertFrom-Json | ForEach-Object { $PSItem.title }
+$ProgressPreference = "SilentlyContinue"
+$requestParameters = @{
+    Uri = "https://api.github.com/repos/$Repository/pulls?state=open&per_page=100"
+    Method = "Get"
+    Headers = @{ Accept = "application/vnd.github.v3+json" }
+}
+$titles = (Invoke-WebRequest @requestParameters).Content | ConvertFrom-Json | ForEach-Object { $PSItem.title }
```
Comment (BenedekFarkas) [inline .github/actions/verify-submodule-pull-request/Check-Parent.ps1:11]: Confirmed that works as expected, but the "real-world" execution is always on the default branch of the repository, right?

Also, what are the expected error outcomes? If the request fails (e.g., authentication, issue not found), then the script throws an InvalidOperationException and displays the 404 status code:
```
Invoke-WebRequest : The remote server returned an error: (404) Not Found.
At [...].ps1:39 char:1
+ Invoke-WebRequest @requestParameters
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-WebRequest], WebExc
   eption
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand
```

File: .github/actions/create-jira-issues-for-community-activities/action.yml [hunk 13]
```diff
@@ -12,8 +12,8 @@ inputs:
     required: false
     default: "true"
     description: >
-        If set to "true", Jira issue titles will be suffixed with "in <issue-component>" if the issue-component
-        parameter is specified, or the repository name. If set to "false", no suffix will be added to Jira issue titles.
+      If set to "true", Jira issue titles will be suffixed with "in <issue-component>" if the issue-component
+      parameter is specified, or the repository name. If set to "false", no suffix will be added to Jira issue titles.
```
Comment (BenedekFarkas) [inline .github/actions/create-jira-issues-for-community-activities/action.yml:16]: This is VS Code autoformatting with the GHA extension.

File: .github/actions/create-jira-issues-for-community-activities/action.yml [hunk 14]
```diff
@@ -126,4 +130,4 @@ runs:
         issue-number: ${{ github.event.issue.number }}
         title: ${{ github.event.issue.title }} (${{ steps.create-issue.outputs.issue }})
         body: "${{ github.event.issue.body }}\n\n[Jira issue](${{ env.JIRA_BASE_URL }}/browse/${{ steps.create-issue.outputs.issue }})"
-        update-mode: 'replace'
+        update-mode: "replace"
```
Comment (BenedekFarkas) [inline .github/actions/create-jira-issues-for-community-activities/action.yml:134]: Also autoformatting.

File: .github/actions/create-jira-issues-for-community-activities/action.yml [hunk 15]
```diff
@@ -100,21 +100,25 @@ runs:
...
+            ContentType = 'application/json'
+            Headers = @{
+                Accept = 'application/json'
+                Authorization = 'Basic ' + [Convert]::ToBase64String([Text.Encoding]::UTF8.GetBytes('${{ env.JIRA_USER_EMAIL }}:${{ env.JIRA_API_TOKEN }}'))
+                ContentType = 'application/json'
+            }
+            Body = ConvertTo-Json($linkProperties)
+        }
+
+        Invoke-WebRequest @requestParameters
```
Comment (BenedekFarkas) [inline .github/actions/create-jira-issues-for-community-activities/action.yml:122]: Replaced curl with Invoke-WebRequest here too, there are no more occurrences in this repo.

File: .github/actions/add-jira-issue-code-to-pull-request/AddJiraIssueCodeToPullRequest.ps1 [hunk 16]
```diff
@@ -13,30 +12,36 @@ $jiraBrowseUrl = $JiraBaseUrl + '/browse/';
...
 }
 
-if (($Title -ne $originalTitle) -or ($Body -ne $originalBody)) {
-    $bodyParams = @{"title" = $Title; "body" = $Body} | ConvertTo-Json
+if (($Title -ne $originalTitle) -or ($Body -ne $originalBody))
+{
+    $bodyParams = @{"title" = $Title; "body" = $Body } | ConvertTo-Json
     $url = "https://api.github.com/repos/$GitHubRepository/pulls/$PullRequestId"
     Invoke-WebRequest $url -Headers (Get-GitHubApiAuthorizationHeader) -Method Patch -Body $bodyParams
 }
```
Comment (BenedekFarkas) [inline .github/actions/add-jira-issue-code-to-pull-request/AddJiraIssueCodeToPullRequest.ps1:47]: If I remove the issue link from the PR description and re-run the corresponding workflow job, was it supposed to re-add it again or only when the PR is being created?

File: .github/actions/verify-submodule-pull-request/Check-Parent.ps1 (hunk 1 above)
Comment (BenedekFarkas) [reply .github/actions/verify-submodule-pull-request/Check-Parent.ps1]: These actions/workflows were executing according to the issue branch in the last run, still all green.

**Comment by BenedekFarkas** [comment]
Does OSOCE execute PS analysis at all? PS Core analysis is fine locally, but Windows PS throws a ton of analyzer errors for LGHA scripts. IMO it should comply with the same rules as the other repositories where we have PS analysis (Utility and Infrastructure Scripts).

**Comment by Piedone** [comment]
It does, and it does with the same config. There doesn't supposed to be, but might be a difference due to in OSOCE it executing during .NET build, and the other two repos you mention directly by running _Invoke-Analyzer.ps1_.

**Comment by BenedekFarkas** [comment]
I see, well, that's not set up correctly then. `Lombiq.Analyzers.PowerShell.targets` only runs pwsh (if available), powershell is only used as a fallback. EZ fix though.

**Comment by Piedone** [comment]
Oh wait, you're talking about pwsh vs Windows PowerShell. Then it's deliberate that we don't care about Windows PowerShell, the more recent and cross-platform pwsh is what we need to use.

**Comment by BenedekFarkas** [comment]
Gotcha!

File: .github/actions/add-jira-issue-code-to-pull-request/AddJiraIssueCodeToPullRequest.ps1 (hunk 16 above)
Comment (Piedone) [reply .github/actions/add-jira-issue-code-to-pull-request/AddJiraIssueCodeToPullRequest.ps1:47]: While this is a nice side effect, I don't think we intended this to actually run on all pushes, it just happens to do without any further checks. But an `if` on the GH even to see if it's a PR `opened` event would be great.

File: .github/actions/create-jira-issues-for-community-activities/action.yml [hunk 17]
```diff
@@ -100,21 +100,25 @@ runs:
...
-          --header 'Accept: application/json' `
-          --header 'Content-Type: application/json' `
-          --fail `
-          --data '{
-                \"object\": {
-                    \"url\":\"${{ steps.issue-details.outputs.link-url }}\",
-                    \"title\":\"${{ steps.issue-details.outputs.link-title }}\"
-                }
-            }'
+        $linkProperties = @{
```
Comment (Piedone) [inline .github/actions/create-jira-issues-for-community-activities/action.yml:103]: Please test this in a throwaway test repo.

File: .github/actions/create-jira-issues-for-community-activities/action.yml [hunk 18]
```diff
@@ -100,21 +100,25 @@ runs:
...
+                title = '${{ steps.issue-details.outputs.link-title }}'
+            }
+        }
+        $requestParameters = @{
+            Uri = '${{ env.JIRA_BASE_URL }}/rest/api/3/issue/${{ steps.create-issue.outputs.issue }}/remotelink'
+            Method = 'POST'
+            ContentType = 'application/json'
+            Headers = @{
+                Accept = 'application/json'
+                Authorization = 'Basic ' + [Convert]::ToBase64String([Text.Encoding]::UTF8.GetBytes('${{ env.JIRA_USER_EMAIL }}:${{ env.JIRA_API_TOKEN }}'))
```
Comment (Piedone) [inline .github/actions/create-jira-issues-for-community-activities/action.yml]: Save `[Convert]` to a variable so that and this line can be shorter.

File: .github/actions/verify-submodule-pull-request/Check-Parent.ps1 (hunk 12 above)
Comment (Piedone) [reply .github/actions/verify-submodule-pull-request/Check-Parent.ps1:11]: This always runs for a PR, so this is fine.

No non-200 responses are expected here.

File: PSScriptAnalyzerSettings.psd1 [hunk 19]
```diff
@@ -1,7 +0,0 @@
-@{
```
Comment (Piedone) [inline PSScriptAnalyzerSettings.psd1:1]: Why was this deleted?

**Review by Piedone — CHANGES_REQUESTED** [review]

File: .github/actions/create-jira-issues-for-community-activities/action.yml (hunk 15 above)
Comment (Piedone) [inline .github/actions/create-jira-issues-for-community-activities/action.yml:122]: When this fails (e.g. the Jira user doesn't have access to the given project, or if the base URL is wrong), will it display the response output? If yes, then we can close https://github.com/Lombiq/GitHub-Actions/issues/53.

File: .github/actions/add-jira-issue-code-to-pull-request/AddJiraIssueCodeToPullRequest.ps1 (hunk 16 above)
Comment (BenedekFarkas) [reply .github/actions/add-jira-issue-code-to-pull-request/AddJiraIssueCodeToPullRequest.ps1:47]: OK, I don't think that's a problem overall, but let's not add that to this issue anyway, because there's already a lot of scope creep.

File: .github/actions/create-jira-issues-for-community-activities/action.yml (hunk 15 above)
Comment (BenedekFarkas) [reply .github/actions/create-jira-issues-for-community-activities/action.yml:122]: Here are the responses:

Wrong email address or issue key doesn't exist:
```
Invoke-WebRequest: [...].ps1:20:13
Line |
  20 |  $response = Invoke-WebRequest @requestParameters
     |              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | {"errorMessages":["Issue does not exist or you do not have permission to see it."],"errors":{}}
```

Wrong access token (probably it thinks that if it's not a valid token, then it's a password?):
```
Invoke-WebRequest: [...].ps1:20:1
Line |
  20 |  Invoke-WebRequest @requestParameters
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Basic authentication with passwords is deprecated.  For more information, see: https://developer.atlassian.com/cloud/confluence/deprecation-notice-basic-auth/ 
```

Link title missing:
```
Invoke-WebRequest: [...].ps1:20:1
Line |
  20 |  Invoke-WebRequest @requestParameters
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | {"errorMessages":[],"errors":{"title":"'Link Text' is required."}}
```

Link URL missing:
```
Invoke-WebRequest: [...].ps1:20:1
Line |
  20 |  Invoke-WebRequest @requestParameters
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | {"errorMessages":[],"errors":{"url":"'URL' is required."}}
```

Success prints out the response details, same as other scripts, so everything's covered!

File: .github/actions/create-jira-issues-for-community-activities/action.yml (hunk 18 above)
Comment (BenedekFarkas) [reply .github/actions/create-jira-issues-for-community-activities/action.yml]: Not sure what you mean by that, but I split it up along method calls to make it more readable.

File: .github/actions/verify-submodule-pull-request/Check-Parent.ps1 (hunk 12 above)
Comment (BenedekFarkas) [reply .github/actions/verify-submodule-pull-request/Check-Parent.ps1:11]: I know, what I was asking is how errors should be handled here. For example:

`Check-Parent.ps1 -Repository "asdf/qwer"` returns
```
Invoke-WebRequest: [...]\Check-Parent.ps1:12:12
Line |
  12 |  $titles = (Invoke-WebRequest @requestParameters).Content | ConvertFro …
     |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | {"message":"Not Found","documentation_url":"https://docs.github.com/rest/reference/pulls#list-pull-requests"}
```

and an invalid branch, like `Check-Parent.ps1 -Repository "Lombiq/GitHub-Actions" -Branch "asdf"` outputs nothing, because the script exits at line 16. If those are OK/expected, then it's good to go.

File: PSScriptAnalyzerSettings.psd1 (hunk 19 above)
Comment (BenedekFarkas) [reply PSScriptAnalyzerSettings.psd1:1]: This should've been removed during https://github.com/Lombiq/PowerShell-Analyzers/issues/12, so I added the explanation in the commit message.

File: .github/actions/create-jira-issues-for-community-activities/action.yml (hunk 17 above)
Comment (BenedekFarkas) [reply .github/actions/create-jira-issues-for-community-activities/action.yml:103]: Done: https://github.com/Lombiq/OSOE-482-Test/issues/2
See the remote link title in the Jira issue compared to line 106 below.

File: .github/actions/create-jira-issues-for-community-activities/action.yml [hunk 20]
```diff
@@ -100,21 +100,26 @@ runs:
...
-          --data '{
-                \"object\": {
-                    \"url\":\"${{ steps.issue-details.outputs.link-url }}\",
-                    \"title\":\"${{ steps.issue-details.outputs.link-title }}\"
-                }
-            }'
+        $linkProperties = @{
+            object = @{
+                url = '${{ steps.issue-details.outputs.link-url }}'
+                title = '${{ steps.issue-details.outputs.link-title }} opened with Invoke-WebRequest'
```
Comment (BenedekFarkas) [inline .github/actions/create-jira-issues-for-community-activities/action.yml]: ` opened with Invoke-WebRequest` to be removed from the title!

File: .github/actions/build-dotnet/Build-DotNetSolutionOrProject.ps1 (hunk 2 above)
Comment (BenedekFarkas) [reply .github/actions/build-dotnet/Build-DotNetSolutionOrProject.ps1]: I've reverted this change, since LGHA doesn't run analysis on Windows PS, only PS Core, so `||` is allowed. If this needs change/improvement, please move that to a new issue.

File: .github/actions/add-jira-issue-code-to-pull-request/AddJiraIssueCodeToPullRequest.ps1 (hunk 16 above)
Comment (Piedone) [reply .github/actions/add-jira-issue-code-to-pull-request/AddJiraIssueCodeToPullRequest.ps1:47]: Added https://github.com/Lombiq/GitHub-Actions/issues/152 for this.

File: .github/actions/verify-submodule-pull-request/Check-Parent.ps1 [hunk 21]
```diff
@@ -3,12 +3,17 @@
     [string] $Branch
 )
 
-$url = "https://api.github.com/repos/$Repository/pulls?state=open&per_page=100"
-$titles = curl -s -H 'Accept: application/vnd.github.v3+json' $url | ConvertFrom-Json | ForEach-Object { $PSItem.title }
+$ProgressPreference = "SilentlyContinue"
```
Comment (Piedone) [inline .github/actions/verify-submodule-pull-request/Check-Parent.ps1]: This isn't used.

File: .github/actions/verify-submodule-pull-request/Check-Parent.ps1 (hunk 12 above)
Comment (Piedone) [reply .github/actions/verify-submodule-pull-request/Check-Parent.ps1:11]: That's OK.

File: .github/actions/create-jira-issues-for-community-activities/action.yml (hunk 20 above)
Comment (Piedone) [reply .github/actions/create-jira-issues-for-community-activities/action.yml]: Looks good!

File: .github/actions/create-jira-issues-for-community-activities/action.yml (hunk 15 above)
Comment (Piedone) [reply .github/actions/create-jira-issues-for-community-activities/action.yml:122]: Looks good, we don't need https://github.com/Lombiq/GitHub-Actions/issues/53, then.

**Review by Piedone — CHANGES_REQUESTED** [review]
GH sent my comments before I submitted the review somehow, but see the inline comments.

File: .github/actions/verify-submodule-pull-request/Check-Parent.ps1 (hunk 21 above)
Comment (BenedekFarkas) [reply .github/actions/verify-submodule-pull-request/Check-Parent.ps1]: It's a global override for the current session not to display a progress bar (while the request is executing). It's recommended for automated sessions, but I just checked, and it doesn't look like we use it anywhere else, so GHA doesn't care - I'll remove it.

File: .github/actions/create-jira-issues-for-community-activities/action.yml (hunk 20 above)
Comment (BenedekFarkas) [reply .github/actions/create-jira-issues-for-community-activities/action.yml]: Reverted and deleted test issues and repo.

**Comment by BenedekFarkas** [comment]
@Piedone please check the last two commits and remaining conversations.

**Review by Piedone — APPROVED** [review]
